dashboard/app/models/user.rb

```diff
@@ -2320,6 +2326,13 @@ def update_share_setting
     return true
   end
 
+  # Updates the child_account_compliance_state attribute to the given state.
+  # @param {String} new_state - A constant from User::ChildAccountCompliance
+  def update_child_account_compliance(new_state)
```

Comment:
`assign` is more correct because `#save` is not called there.
```suggestion
  def assign_child_account_compliance(new_state)
```

Reply:
I'd generally agree if not for the other methods following such a convention. I'll take a look... calling save isn't out if the question either.

Reply:
@wilkie It's not about the naming convention but about the misleading behind method naming.

`#update_` means that some data will be saved in DB and we can see it in most of the `User#update_`methods:
> https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L341
> https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L930

But we also have methods with incorrect naming. In this case, we should use `#assign_` instead of `#update_` because the data is set without saving/updating: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L2335